OpenRA.Game/Network/ReplayConnection.cs

```diff
@@ -55,6 +56,14 @@ public ReplayConnection(string replayFilename)
 					var frame = BitConverter.ToInt32(packet, 0);
 					chunk.Packets.Add(Pair.New(client, packet));
 
+                    if (frame != int.MaxValue)
+                    {
+                        if (!lastClientsFrame.ContainsKey(client))
+                            lastClientsFrame[client] = frame;
+                        else if (frame > lastClientsFrame[client])
+                            lastClientsFrame[client] = frame;
```

Comment:
Since both these branches do the same thing, you should probably combine the condition into one statement. i.e. ` if (!lastClientsFrame.ContainsKey(client) || frame > lastClientsFrame[client])`